### PR TITLE
Fixed EntryLoaderInitializationTest (#15376)

### PR DESCRIPTION
**Pull request by vojtechtoman**
Made `test[Eager|Lazy]EntryLoaderInitializesValues_withExpirationTime`
not sensitive to `GetOperation` retries.

Fixes https://github.com/hazelcast/hazelcast/issues/15376

### CODE REVIEW

**Comment by vojtechtoman** [comment]
run-lab-run

**Comment by vojtechtoman** [comment]
run-lab-run

**Comment by vojtechtoman** [comment]
run-lab-run

**Comment by vojtechtoman** [comment]
run-lab-run

**Comment by vojtechtoman** [comment]
Modified `TestEntryLoader` to use `ConcurrentHashMap` instead of regular `HashMap` for the records storage. There was no happens-before relationship between inserting records (using `putExternally()`) and subsequent reads of the records from different threads during entry loading. As a result, the loading threads may have seem incomplete records-set. This in turn may have resulted in subsequent (and unexpected in the test) get operations.

Another possible scenario for unexpected gets might be (non-retryable) exceptions during `LoadAllOperations` but there is no evidence of those in the logs.

File: hazelcast/src/test/java/com/hazelcast/map/impl/mapstore/TestEntryLoader.java [hunk 1]
```diff
@@ -21,22 +21,26 @@
...
 public class TestEntryLoader implements EntryLoader<String, String> {
 
     static final String NULL_RETURNING_KEY = "nullReturningKey";
 
-    private Map<String, Record> records = new HashMap<>();
+    private Map<String, Record> records = new ConcurrentHashMap<>();
     private AtomicInteger loadedEntryCount = new AtomicInteger();
     private AtomicInteger loadAllCallCount = new AtomicInteger();
     private AtomicInteger loadCallCount = new AtomicInteger();
+    private Set<String> loadUniqueKeys = ConcurrentHashMap.newKeySet();
```
Comment (petrpleshachkov) [inline hazelcast/src/test/java/com/hazelcast/map/impl/mapstore/TestEntryLoader.java]: Minor: we can make this field `final` (as well as other fields in this class).

Comment (vojtechtoman) [reply hazelcast/src/test/java/com/hazelcast/map/impl/mapstore/TestEntryLoader.java]: Made the fields final.

**Pull request merged**